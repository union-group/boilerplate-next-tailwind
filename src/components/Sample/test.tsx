import { render, screen } from '@testing-library/react'

import { Sample } from '.'

const { getByRole } = screen

describe('<Button />', () => {
  it('should render by default', () => {
    render(<Sample>Children</Sample>)

    const component = getByRole('button', { name: 'Children' })

    expect(component).toHaveClass('bg-primary text-zinc-50 py-4 w-[21.188rem]')
  })

  it('should render with outline variant', () => {
    render(<Sample variant="outline">Children</Sample>)

    const component = getByRole('button', { name: 'Children' })

    expect(component).toHaveClass(
      'bg-transparent border-2 border-primary text-primary py-4 w-[21.188rem]',
    )
  })

  it('should render with correct sizes', () => {
    const { rerender } = render(<Sample size="small">Children</Sample>)

    const component = getByRole('button', { name: 'Children' })

    expect(component).toHaveClass('py-0 w-[16.188rem]')

    rerender(<Sample size="medium">Children</Sample>)

    expect(component).toHaveClass('py-4 w-[21.188rem]')

    rerender(<Sample size="large">Children</Sample>)

    expect(component).toHaveClass('w-[22.188rem]')
  })

  it('should match snapshot', () => {
    const { container } = render(<Sample>Children</Sample>)

    expect(container).toMatchSnapshot()
  })
})
