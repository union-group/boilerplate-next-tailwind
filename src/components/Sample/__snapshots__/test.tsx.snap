// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match snapshot 1`] = `
<div>
  <button
    class="flex h-12 items-center justify-center rounded-10 text-lg font-bold  bg-primary text-zinc-50 py-4 w-[21.188rem]"
  >
    Children
  </button>
</div>
`;
